src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.cs

```diff
@@ -46,6 +46,8 @@ private struct CacheLineSeparated
         private static int s_completionCount = 0;
         private static int s_threadAdjustmentInterval;
 
+        private static volatile int s_numRequestedWorkers = 0;
+
         static ClrThreadPool()
```

Comment:
I think so.

An alternative that tends to work pretty well in C# is to have just a single static that points to a singleton instance, and have all state as regular instance fields instead.